core/src/mindustry/logic/LExecutor.java

```diff
@@ -1980,8 +1992,15 @@ public void run(LExecutor exec){
                 }else if(type == LMarkerControl.flushText){
                     marker.setText(exec.textBuffer.toString(), true);
                     exec.textBuffer.setLength(0);
+                }else if(type == LMarkerControl.texture){
+                    if(exec.obj(p1) != null){
+                        StringBuilder res = new StringBuilder(exec.obj(p1).toString());
```

Comment:
This implementation is pretty weird. I don't think it should add `-` between images - what if someone wants to draw `stone1, stone2, stone3`?

Additionally, it should check for `p2` && `p3` being null, and avoid allocating the StringBuilder in that case.

Reply:
IMO, `stone1`, `stone2` and `stone3` are themselves inconsistent. What if you wanted to have an `@type` variant drawn, but all of your code is written for conveyor rotations, animation frames and (rarely) shading?

Reply:
Check the new texture implementation, you can now flush print buffer as texture name in `setmarker`. This solves the problem, i guess? Or are you talking about something else?